src/dhcpmon/src/dhcp_mon.c

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file dhcp_mon.c
+ *
+ *  @brief dhcp relay monitor module
+ */
+
+#include <signal.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <syslog.h>
+#include <assert.h>
+
+#include "dhcp_mon.h"
+#include "dhcp_devman.h"
+
+/** window_interval_sec monitoring window for dhcp relay health checks */
+static int window_interval_sec = 12;
+/** dhcp_unhealthy_max_count max count of consecutive unhealthy statuses before reporting to syslog */
+static int dhcp_unhealthy_max_count = 10;
+/** dhcp_mon_active control monitoring thread main loop */
+static int dhcp_mon_active = 0;
+/** libevent base struct */
+static struct event_base *base;
+/** libevent timeout event struct */
+static struct event *ev_timeout = NULL;
+/** libevent signal event struct */
+static struct event *ev_sigint;
+
+/**
+ * @code signal_callback(fd, event, arg);
+ *
+ * @brief signal handler for dhcpmon. It will initiate shutdown when signal is caught
+ *
+ * @param fd        libevent socket
+ * @param event     event triggered
+ * @param arg       pointer user provided context (libevent base)
+ *
+ * @return none
+ */
+static void signal_callback(evutil_socket_t fd, short event, void *arg)
+{
+    syslog(LOG_ALERT, "Received signal %d\n", event);
+    dhcp_devman_print_status();
+    dhcp_mon_stop();
+}
+
+/**
+ * @code timeout_callback(fd, event, arg);
+ *
+ * @brief periodic timer call back
+ *
+ * @param fd        libevent socket
+ * @param event     event triggered
+ * @param arg       pointer user provided context (libevent base)
+ *
+ * @return none
+ */
+static void timeout_callback(evutil_socket_t fd, short event, void *arg)
+{
+    static int count = 0;
+    dhcp_mon_status_t dhcp_mon_status = dhcp_devman_get_status();
+
+    switch (dhcp_mon_status)
+    {
+    case DHCP_MON_STATUS_UNHEALTHY:
+        if (++count > dhcp_unhealthy_max_count) {
+            syslog(LOG_ALERT, "DHCP Relay is not healthy after %d health checks\n", count);
+        }
+        break;
+    case DHCP_MON_STATUS_HEALTHY:
+        if (count > 0) {
+//            syslog(LOG_INFO, "DHCP Relay became healthy after %d health checks\n", count);
```

Comment:
remove it?

Reply:
I was thinking about having filtering in our syslog so we can debug without having to recompile the binary. So, I left those commented until filtering is implemented and I could uncomment them. I do not feel strongly about leaving them behind though.